lib/component.js

```diff
@@ -161,6 +161,21 @@ class Component extends WebComponent {
     }
   }
 
+  /**
+   * Returns the reference to the parent of the given name. An error is thrown out if no parent of
+   * such name is found.
+   * @param {tagName} - name of the parent to search for.  @example
+   * this.parentbyTagName('myApp');
+   */
+  parentByTagName(tagName) {
+    for (let node = this; node; node = node.parentNode) {
```

Comment:
This looks like a generic DOM-traversal helper. Are you sure you're not looking for something to traverse the custom component hierarchy as your example seems to imply? i.e. `node = node.$panelParent;`